### PR TITLE
delay type completion to improve editor response time

### PR DESCRIPTION
**Pull request by katepol**

### CODE REVIEW

File: cell/src/test/java/jetbrains/jetpad/cell/text/TextEditingTest.java [hunk 1]
```diff
@@ -46,9 +47,12 @@
   private TextCell navTextView2 = new TextCell();
 
   private String completed;
+  private TestEventDispatchThread edt = new TestEventDispatchThread();
```
Comment (timzam) [inline cell/src/test/java/jetbrains/jetpad/cell/text/TextEditingTest.java]: Minor. The edt field could be final here and in other tests.

File: hybrid/src/test/java/jetbrains/jetpad/hybrid/BaseHybridEditorEditingTest.java [hunk 2]
```diff
@@ -66,6 +99,7 @@
   protected Cell targetCell;
 
   private Registration registration;
+  private TestEventDispatchThread edt = new TestEventDispatchThread();
```
Comment (timzam) [inline hybrid/src/test/java/jetbrains/jetpad/hybrid/BaseHybridEditorEditingTest.java]: The field could be final.

File: hybrid/src/test/java/jetbrains/jetpad/hybrid/HybridEditorWithTransformedListTest.java [hunk 3]
```diff
@@ -36,6 +45,7 @@
   private ExprList exprList = new ExprList();
   private Registration registration;
   private ExprListMapper rootMapper = new ExprListMapper(exprList);
+  private TestEventDispatchThread edt = new TestEventDispatchThread();
```
Comment (timzam) [inline hybrid/src/test/java/jetbrains/jetpad/hybrid/HybridEditorWithTransformedListTest.java]: The field could be final.

File: projectional/src/test/java/jetbrains/jetpad/projectional/cell/SideTransformTest.java [hunk 4]
```diff
@@ -15,41 +15,55 @@
...
+import jetbrains.jetpad.projectional.testApp.model.IdExpr;
+import jetbrains.jetpad.projectional.testApp.model.PlusExpr;
 import org.junit.Before;
 import org.junit.Test;
 
 import static org.junit.Assert.assertSame;
 import static org.junit.Assert.assertTrue;
 
 public class SideTransformTest extends EditingTestCase {
+  private TestEventDispatchThread edt = new TestEventDispatchThread();
```
Comment (timzam) [inline projectional/src/test/java/jetbrains/jetpad/projectional/cell/SideTransformTest.java]: The field could be final.

File: cell/src/main/java/jetbrains/jetpad/cell/util/DebouncedCommand.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  public void dispose() {
+    removeRegistration();
+  }
+
+  private void removeRegistration() {
+    if (myRegistration != null) {
+      myRegistration.remove();
+    }
+  }
+}
```
Comment (timzam) [inline cell/src/main/java/jetbrains/jetpad/cell/util/DebouncedCommand.java]: No newline at the end of the file.

Comment (katepol) [inline cell/src/main/java/jetbrains/jetpad/cell/util/DebouncedCommand.java]: Why should it be there?

Comment (timzam) [inline cell/src/main/java/jetbrains/jetpad/cell/util/DebouncedCommand.java]: "We need to include empty line at end of file. Missing it gives warning by default in git UIs including github."

**Comment by katepol** [comment]
Done

**Pull request merged**